src/site/markdown/getting-started.md

```diff
@@ -0,0 +1,94 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Getting Started
+
+## Overview
+
+Using the `Verifier` consists out of three different phases
+
+1. Configure
+2. Run
+3. Verify
+
+## Configure
+
+The `Verifier` instance is constructed in the simplest case with just one argument taking the path name of the base directory containing the `pom.xml` of the project you want to build.
+
+```
+String baseDir = ...
+Verifier verifier = new Verifier(baseDir);
+```
+
+The configuration can be further tweaked with additional setter methods and/or constructors taking more parameters.
+
+### System Properties
+
+The following systen properties/environment variables are evaluated
+
+| System Property | Description |
+| --- | --- |
+| `verifier.forkMode` | The following values are supported: <br/>`auto` uses the forked launcher when environment variables are set<br/>`embedder` always uses the embedded launcher<br/>any other value leads to always using the forked launcher
+| `maven.home` | The directory containing the Maven executable in `bin/mvn` |
+| `user.home` | Set by JRE, used for determining Maven default local repository path or the fallback Maven executable |
+| `maven.bootclasspath` | Only relevant if Maven home could be determined and the embedded launcher is being used. Determines the classpath of the launcher. May contain multiple path separated by the system dependent path separator.
+| `classworlds.conf` | Only relevant if Maven home could be determined and the embedded launcher is being used. The configuration file used by [Plexus Classworlds Loader][plexus-classwords]. If not set defaults to `<Maven Home>/bin/m2.conf`. |
+| `maven.repo.local` | Contains the path of the local Maven repository |
+| `maven.repo.local.layout` | Layout of the local Maven repository. Either `legacy` or `default` with the latter being the default. |
+
+
+### Finding Maven Executable
+
+The following mechanism determines the binary which is used for executing Maven. This is 
+
+- either the embedded launcher which uses
+    -  the `org.apache.maven.cli.MavenCli` class loaded from the context class loader (in case Maven Home could not be determined) or 
+    -  the [Plexus Classworlds Loader][plexus-classwords] (in case Maven Home could be determined)
+-  or the forked launcher
+
+Whether the embedded or the forked launcher are used depends on the field `forkJvm` set through the constructor or `setForkJvm` or as fallback on the value of system property `verifier.forkMode`.
+
+#### Determining Maven Home directory
+
+The following directories are considered as potential Maven home directory (relevant for both forked launcher and embedded launcher with  [Plexus Classworlds Loader][plexus-classwords]). The first existing directory terminates the mechanism. 
+
+1. Maven Home path given in the constructor
+2. System property `maven.home`
+3. Environment variable `M2_HOME`
```

Comment:
M2_HOME is deprecated - should be removed from code

Reply:
Please create a dedicated JIRA issue for that. This PR is just about documenting the status quo.